courseware/api.py

```diff
@@ -112,6 +115,27 @@ def create_edx_user(user):
         )
         # edX responds with 200 on success, not 201
         if resp.status_code != status.HTTP_200_OK:
+            if settings.OPENEDX_SERVICE_WORKER_API_TOKEN is None:
+                raise CoursewareUserCreateError(
+                    f"Error creating Open edX user. {get_error_response_summary(resp)}"
+                )
+            req_session.headers.update(
+                {
+                    AUTH_TOKEN_HEADER_NAME: "Bearer {}".format(
+                        settings.OPENEDX_SERVICE_WORKER_API_TOKEN
+                    ),
+                    API_KEY_HEADER_NAME: settings.OPENEDX_API_KEY,
+                }
+            )
+            get_response = req_session.get(
+                edx_url(f"{OPENEDX_USER_ACCOUNT_DETAIL_PATH}/{user.username}")
+            )
+            # user already exists on edx
+            if (
+                get_response.status_code == status.HTTP_200_OK
+                and get_response.json().get("email") == user.email
+            ):
```

Comment:
Could you split this out into a separate function? Both for cleaner code and because I don't think it should be reusing the same requests session.